packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -1,20 +1,340 @@
+        );
+        if (singleSourceSamplesWithOutput.length < MIN_NUM_SAMPLE_INPUTS) {
+            continue;
+        }
+
+        // TODO(kimpers): Do we need to handle 0 entries, from eg Kyber?
+        const serializedPath = singleSourceSamplesWithOutput.reduce<SerializedPath>(
+            (memo, sample, sampleIdx) => {
+                memo.ids.push(sampleToId(sample.source, sampleIdx));
+                memo.inputs.push(sample.input.integerValue().toNumber());
+                memo.outputs.push(sample.output.integerValue().toNumber());
+                memo.outputFees.push(
+                    calculateOuputFee(side, sample, opts.outputAmountPerEth, opts.inputAmountPerEth, fees)
+                        .integerValue()
+                        .toNumber(),
+                );
+
+                return memo;
+            },
+            {
+                ids: [],
+                inputs: [],
+                outputs: [],
+                outputFees: [],
+            },
+        );
+
+        samplesAndNativeOrdersWithResults.push(singleSourceSamplesWithOutput);
+        serializedPaths.push(serializedPath);
+    }
+
+    for (const [idx, nativeOrder] of nativeOrders.entries()) {
+        const { input: normalizedOrderInput, output: normalizedOrderOutput } = nativeOrderToNormalizedAmounts(
+            side,
+            nativeOrder,
+        );
+        // NOTE: skip dummy order created in swap_quoter
+        // TODO: remove dummy order and this logic once we don't need the JS router
+        if (normalizedOrderInput.isLessThanOrEqualTo(0) && normalizedOrderOutput.isLessThanOrEqualTo(0)) {
+            continue;
+        }
+
+        // HACK: the router requires at minimum 3 samples as a basis for interpolation
+        const inputs = [
+            0,
+            normalizedOrderInput
+                .dividedBy(2)
+                .integerValue()
+                .toNumber(),
+            normalizedOrderInput.integerValue().toNumber(),
+        ];
+        const outputs = [
+            0,
+            normalizedOrderOutput
+                .dividedBy(2)
+                .integerValue()
+                .toNumber(),
+            normalizedOrderOutput.integerValue().toNumber(),
+        ];
+        // NOTE: same fee no matter if full or partial fill
+        const fee = calculateOuputFee(side, nativeOrder, opts.outputAmountPerEth, opts.inputAmountPerEth, fees)
+            .integerValue()
+            .toNumber();
+        const outputFees = [fee, fee, fee];
+        // NOTE: ids can be the same for all fake samples
+        const id = sampleToId(ERC20BridgeSource.Native, idx);
+        const ids = [id, id, id];
+
+        const serializedPath: SerializedPath = {
+            ids,
+            inputs,
+            outputs,
+            outputFees,
+        };
+
+        samplesAndNativeOrdersWithResults.push([nativeOrder]);
+        serializedPaths.push(serializedPath);
+    }
+
+    const rustArgs: OptimizerCapture = {
```

Comment:
Oh I did not expect you guys to keep using my data dump hack input types and names.

Reply:
Is that why we're still passing in path IDs?